vhost-device-sound/src/device.rs

```diff
@@ -155,137 +155,89 @@ impl VhostUserSoundThread {
         let mut any = false;
 
         for desc_chain in requests {
-            let descriptors: Vec<_> = desc_chain.clone().collect();
-            if descriptors.len() < 2 {
-                return Err(Error::UnexpectedDescriptorCount(descriptors.len()).into());
-            }
+            let mem = atomic_mem.memory();
 
-            // Request descriptor.
-            let desc_request = descriptors[0];
-            if desc_request.is_write_only() {
-                return Err(Error::UnexpectedWriteOnlyDescriptor(0).into());
-            }
+            let Ok(mut reader) = desc_chain.clone().reader(&mem) else {
+                return Err(Error::DescriptorReadFailed.into());
+            };
```

Comment:
The else branch could be a `.map_err(...)?`, right? Maybe it would make sense to bubble up the internal error too by wrapping it in `DescriptorReadFailed`

same for later code